src/main/java/org/fusesource/jansi/internal/musl/AnsiConsoleSupportImpl.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (C) 2009-2023 the original author(s).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fusesource.jansi.internal.musl;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.TimeUnit;
+
+import org.fusesource.jansi.internal.AnsiConsoleSupport;
+
+import static java.nio.charset.StandardCharsets.US_ASCII;
+
+public final class AnsiConsoleSupportImpl implements AnsiConsoleSupport {
+
+    @Override
+    public String getProviderName() {
+        return "musl";
+    }
+
+    @Override
+    public CLibrary getCLibrary() {
+        return new CLibrary() {
+            private final String[] command;
+
+            {
+                String stty = "stty";
+
+                String path = System.getenv("PATH");
+                if (path != null) {
+                    for (String p : path.split(File.pathSeparator)) {
+                        Path sttyFile = Paths.get(p, "stty");
+                        if (Files.isExecutable(sttyFile)) {
+                            stty = sttyFile.toAbsolutePath().normalize().toString();
+                        }
+                    }
+                }
+
+                command = new String[] {stty, "size"};
+            }
+
+            @Override
+            public short getTerminalWidth(int fd) {
+                if (isTty(fd) == 0) {
+                    return 0;
+                }
+
+                Process process = null;
+                try {
+                    process = new ProcessBuilder(command)
+                            .redirectInput(ProcessBuilder.Redirect.INHERIT)
```

Comment:
I think that's slightly wrong. The terminal width usually refer to the output rather than the input.  See how it's handled in `MingwSupport` above.  

Btw, we may want to port the change I made earlier in JLine related to this: https://github.com/jline/jline3/pull/868/files

Reply:
I refactored MingwSupport to reuse its.